src/main/java/org/apache/commons/lang3/LocaleUtils.java

```diff
@@ -248,7 +250,9 @@ private static Locale parseLocale(final String str) {
             return new Locale(str);
         }
 
-        final String[] segments = str.split("_", -1);
+        final String[] segments = str.indexOf(DASH) != -1
+            ? str.split(String.valueOf(DASH), -1)
```

Comment:
Does it matter that dash is tested before underscore?

Reply:
It's been a while since I wrote this code, but I don't think so. Should be possible to switch them around without issues.

Reply:
All the unit tests still pass with the comparison swapped. I made the change in 0319d84d1. It's probably the right thing to do to optimize for the previous delimiter as that'll be the vast majority of use-cases.